src/gnatss/loaders.py

```diff
@@ -170,38 +172,65 @@ def load_travel_times(
     return all_travel_times
 
 
-def load_roll_pitch_heading(files: List[str]) -> pd.DataFrame:
+def load_roll_pitch_heading(files: Union[List[str], str]) -> pd.DataFrame:
     """
     Loads roll pitch heading data into a pandas dataframe from a list of files.
 
     Parameters
     ----------
-    files : List[str]
-        The list of path string to files to load
+    files : Union[List[str], str]
+        The list of path string to files to load, or an empty string if no file paths found
```

Comment:
The default value set for the `all_files_dict` is empty string for all the keys values. If no files are found for roll_pitch_heading, quality_controls or deletions, then the key value is an empty string. 
`all_files_dict.setdefault("quality_controls", "")`
`all_files_dict.setdefault("deletions", "")`
`all_files_dict.setdefault("roll_pitch_heading", "")`

The other loaders actually also accept empty strings, so this typehint change should be applied there too.

Reply:
Is that the behavior you're seeing by `gather_files`?

Based on this:
https://github.com/seafloor-geodesy/gnatss/blob/55ff42fea1ff4b206d443f9d1b6a935866a5c0df/src/gnatss/main.py#L53-L63

`path` is an empty string "" if it doesn't exist, see L53

However L61 ensures that it becomes a `List[str]` since we would have `all_files = [""]`

Reply:
Yes. That is the behavior I'm seeing in `gather_files()`. 
For illustration, I removed the `roll_pitch_heading` in the config, and printed out the value of `all_files_dict['roll_pitch_heading']`. As you can see, it is an empty string, and not a list containing an empty string ...
`
all_files_dict['roll_pitch_heading'] =
`

Reply:
Hmm... I'm really not seeing the behavior and that doesn't makes sense at all based on the logic of the code... if `path` variable is any string without "**" it should give a `List[str]`. I tried this locally and I shared you a screenshot of the proper behavior

I changed the path to an empty string for the test config yaml file...
```yaml
posfilter:
  atd_offsets:
    forward: 0.0053
    rightward: 0
    downward: 0.92813
  input_files:
    roll_pitch_heading:
      path: ""
```

Then using the debugger to print out the `all_files_dict` and sure enough it's getting `['']`, which is what I'm expecting


![Screenshot 2024-02-13 at 2 38 07 PM](https://github.com/seafloor-geodesy/gnatss/assets/17802172/b4b8620b-009c-4ea9-b4c5-bdd0bbec4cd7)


So I'm not sure what's going on, on your end to get you an empty string.. anyhow, the input to this function SHOULD NOT be an empty string, but rather a `List[str]`.

Reply:
As discussed in our call, I have made `load_roll_pitch_heading()` files argument accept only List[str].